File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curr=0,max_ans=INT_MIN;
        for(int i=0;i<nums.size();i++)
        {
            curr+=nums[i];
            max_ans=max(curr,max_ans);
            if(curr<0)curr=0;
        }
        return max_ans;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
//kadane most optimal solution 
class Solution {
    public int maxSubArray(int[] nums) {

      int maxSum= nums[0];
      int sum=maxSum>0?maxSum:0;
      for(int i=1;i<nums.length;i++){
        // System.out.println(maxSum +" "+ sum);
        int value=nums[i]+sum;
        if(value>maxSum){
            maxSum=value;
            sum=value;
        }
        else if(value>=0)sum=value;
        if(value<0) sum=0;
      } 
      return maxSum;       
    }
}